contrib/test.sh

```diff
@@ -11,6 +11,11 @@ echo "PATH: \"$PATH\""
 # Pin dependencies for Rust v1.29
 if [ "$TRAVIS_RUST_VERSION" = "1.29.0" ]; then
     cargo generate-lockfile --verbose
+
+    # hyper depends on log 0.3 while we depnd on 0.4, so cargo doesn't know which one to pin
+    LOG_4_PATCH="$(cargo update --package "log" --precise "0.4.13" 2>&1 | sed -n "s/.*log:0.4.\([0-9]*\)/\1/p")"
```

Comment:
I couldn't find a good way to get the log versions so I just ran the command in a way that it fails, giving me the version number to run it correctly.

Reply:
Hmm having added that error in a comment would have been nice, but let's see if this works. I prefer it over the version setting in Cargo.toml.